File: ABC/251-300/277/d.cpp
#include <bits/stdc++.h>
using namespace std;
#define rep(i,n) for(int i=0;i<(n);i++)
using ll = long long;

int main(){
    int n,m;
    cin >> n >> m;
    vector<int> a(n);
    rep(i,n) cin >> a[i];
    map<int,int> mp;
    set<int> st;
    ll sum = 0;
    rep(i,n){
        mp[a[i]]++;
        st.insert(a[i]);
        sum += (ll)a[i];
    }
    sort(a.begin(),a.end());
    int key = a[0];
    set<int> needTovisit;
    needTovisit.insert(key);
    rep(i,n){
        if(key == a[i]) continue;
        if(key + 1 == a[i]) continue;
        needTovisit.insert(a[i]);
        key = a[i];
    }
    if(!needTovisit.count(1) && needTovisit.size() != 1) needTovisit.erase(a[0]);
    auto calc = [&](int x, ll val){
        while(true){
            if(!st.count(x)) break;
            if(x == m-1){
                val -= (ll)mp[x]*x;
                x = 0;
            }else{
                val -= (ll)mp[x]*x;
                x++;
            }
        }
        return val;
    };

    ll ans = sum;
    for(int x : needTovisit){
        cout << x << endl;
        ll now = sum;
        now = calc(x,now);
        ans = min(ans,now);
    }
    cout << ans << endl;
    return 0;
}

File: ABC/251-300/277/d.py
n,m = map(int,input().split())
a = list(map(int,input().split()))
a.sort()
visited = [False for _ in range(n)]
s = 0
for x in a:
    s += x
ans = s
for i in range(n):
    if(visited[i]): continue
    visited[i] = True
    now = 0
    now += a[i]
    j = i
    while(j < n-1):
        if visited[j+1]: break
        if a[j+1] != a[j] and a[j+1] != a[j]+1 : break
        now += a[j+1]
        visited[j+1] = True
        j += 1
    ans = min(ans,s-now)
    
if a[-1] == m-1: #m-1が存在して0につながる場合を考える
    cur = a[0]
    con = False
    #前から消していく
    for i in range(n-1):
        if a[i+1] != a[i] and a[i+1] != a[i]+1:
            if a[i+1] != m-1:
                con = True
            break
        cur += a[i+1]
    #後ろから消していく
    if con:
        a.reverse()
        cur += a[0]
        for i in range(n):
            if a[i+1] != a[i] and a[i+1] != a[i]-1:
                break
            cur += a[i+1]
    ans = min(ans,s-cur)
print(ans)
